### PR TITLE
docs(agents): add breaking schema change and migration policy

### PR DESCRIPTION
**Pull request by rmartz**
AGENTS.md had no guidance on when a Firebase schema change requires a migration. Without this, agents and contributors might rename paths or change field types without providing a migration script, silently corrupting existing user data.

Adds a **Firebase Schema Migrations** section defining:
- What constitutes a breaking change (renamed paths, type changes, removed fields, altered enum values)
- Requirement for an idempotent migration script in `scripts/migrations/` before merging
- Additive changes (new optional fields with safe defaults) are explicitly called out as non-breaking
- Documentation requirement alongside the schema change

---
*Created by Claude Sonnet 4.6*

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -108,6 +108,13 @@ pnpm build-storybook  # Build static Storybook
...
 - **Granularity scales with level of abstraction.** Low-level functions (pure utilities, serializers) warrant thorough edge-case coverage. High-level functions (service orchestration) should have smoke tests that verify they correctly apply the lower-level logic — not re-test every edge case that belongs in the lower-level tests.
 
+## Firebase Schema Migrations
+
+- **Never make a breaking schema change without a migration.** A breaking change is any modification to the Firebase Realtime DB path structure or field format that would cause existing stored data to be misread, ignored, or crash the application — for example: renaming a path segment, changing a field's type, removing a required field, or altering an enum's serialized value.
+- Before merging a breaking schema change, provide a migration script in `scripts/migrations/` that reads the old data shape and writes it in the new shape. The script must be idempotent (safe to run multiple times) and must not delete old data until the migration is verified complete.
```
[the remark this answers is not in the document]
Comment (rmartz) [reply AGENTS.md:114]: Disagreeing here. The point of this policy is to prescribe a canonical location for migration scripts so agents and contributors don't scatter them. Removing the path from the doc makes it *less* actionable, not more. The `scripts/migrations/` directory doesn't need to exist before it's needed — the first migration PR can create it. The doc is intentionally forward-looking and prescriptive.

**Review by rmartz — COMMENTED** [review]
## Overview

Adds a **Firebase Schema Migrations** section to AGENTS.md defining when a breaking schema change requires a migration script. Covers what counts as breaking (renamed paths, type changes, removed required fields, altered enum values), requires an idempotent migration script in `scripts/migrations/`, clarifies additive-only changes are non-breaking, and requires documentation alongside the schema change in `docs/database-schema.md`.

## CI

All checks pass (Build, Format, Lint, Tests, PR title validation, Vercel).

## Open Inline Thread

[AGENTS.md line 114 (copilot-pull-request-reviewer)](https://github.com/rmartz/personal-budget/pull/43#discussion_r3113650263) — Disagreed; replied in thread.

The suggestion removes the specific `scripts/migrations/` path in favor of unspecified guidance. That is the wrong direction — the purpose of this policy is to prescribe a canonical location so agents and contributors don't scatter migration scripts. The directory not yet existing in the repo is fine; the first migration PR creates it. The doc is intentionally forward-looking.

## Code Quality

The content is clear, precise, and consistent with the style of surrounding sections. The distinction between breaking and non-breaking changes is well-drawn with concrete examples. The idempotency and data-safety requirements are appropriate guardrails.

No issues.

## Verdict

All CI checks pass, the PR is docs-only (UAT-exempt), the branch is up to date with main, and the single inline thread has been addressed — ready to merge.

---
*Claude Sonnet 4.6*
